File: tests/rosettacode/hanoi.clj
;; Towers of Hanoi - 4 disks
(defn hanoi [n from to via]
  (when (pos? n)
    (hanoi (dec n) from via to)
    (println (str "Move disk " n " from " from " to " to))
    (hanoi (dec n) via to from)))

(hanoi 4 "A" "C" "B")

File: tests/rosettacode/hanoi.py
# Towers of Hanoi - 4 disks
def hanoi(n, frm, to, via):
    if n == 1:
        print(f"Move disk 1 from {frm} to {to}")
        return
    hanoi(n - 1, frm, via, to)
    print(f"Move disk {n} from {frm} to {to}")
    hanoi(n - 1, via, to, frm)

hanoi(4, "A", "C", "B")
